test/test_find_and_replace.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <iostream>
+
+#include "rcpputils/find_and_replace.hpp"
+
+TEST(test_find_and_replace, find_and_replace) {
+  // Empty input
+  {
+    auto ret = rcpputils::find_and_replace("", "foo", "bar");
+    EXPECT_EQ("", ret);
+  }
+  // Empty find
+  {
+    auto ret = rcpputils::find_and_replace("foo", "", "bar");
+    EXPECT_EQ("foo", ret);
+  }
+  // Empty replace
+  {
+    auto ret = rcpputils::find_and_replace("foo", "foo", "");
+    EXPECT_EQ("", ret);
+  }
+  // Single occurance
+  {
+    auto ret = rcpputils::find_and_replace("foo", "foo", "bar");
+    EXPECT_EQ("bar", ret);
+  }
+  // No occurances
+  {
+    auto ret = rcpputils::find_and_replace("foo", "bar", "baz");
+    EXPECT_EQ("foo", ret);
+  }
+  // Multiple occurances
+  {
+    auto ret = rcpputils::find_and_replace("foobarfoobar", "foo", "baz");
+    EXPECT_EQ("bazbarbazbar", ret);
+  }
+  {
+    auto ret = rcpputils::find_and_replace("foofoobar", "foo", "baz");
+    EXPECT_EQ("bazbazbar", ret);
+  }
+  // find == replace
```

Comment:
Suggestion: This case can be handled separately in the function itself to avoid an unnecessary iteration of the string.

Reply:
Good point. Done in dec7808